.github/workflows/native-mac.yml

```diff
@@ -0,0 +1,120 @@
+name: Native MacOS
+
+on:
+  push:
+    branches: [ main, ci-sandbox ]
+  pull_request:
+    branches: [ main, ci-sandbox ]
+
+jobs:
+  build-macos:
+    strategy: 
+      fail-fast: false
+      matrix:
+        include:
+          - name: "full-marian"
+            os: macos-10.15
+            test_tags: ""
+            cmake: 
+              CMAKE_BUILD_TYPE: "Release"
+              COMPILE_TESTS: "ON"
+              USE_WASM_COMPATIBLE_SOURCE: "OFF"
+              USE_FBGEMM: "OFF"
+              USE_STATIC_LIBS: "OFF"
+              COMPILE_SERVER: "OFF"
+              COMPILE_EXAMPLES: "OFF"
+
+          - name: "full-marian-slim"
+            os: macos-10.15
+            test_tags: ""
+            cmake: 
+              CMAKE_BUILD_TYPE: "Slim"
+              COMPILE_TESTS: "ON"
+              USE_WASM_COMPATIBLE_SOURCE: "OFF"
+              USE_STATIC_LIBS: "OFF"
+              USE_FBGEMM: "OFF"
+              COMPILE_SERVER: "OFF"
+              COMPILE_EXAMPLES: "OFF"
```

Comment:
@jerinphilip So, now there are 4 configs in CI:

`full-marian`
`full-marian-slim`
`minimal-marian`
`wasm`

Do you still want to keep `full-marian-slim` now that you are not uploading any artifacts (because of which this was added in the first place)?

I am fine with keeping it if you have reasons for it. But in this case, CI will be doing 8 checks (instead of 6 if you don't) which will make it slower.

Reply:
CI launches everything in parallel, Slim is faster to check. I don't have much strong opinions here, I just used Slim for faster turnaround cycles while editing only CI. Note that I was doing a lot of trial and error in this PR and 7 mins gain seemed nice at the point :D.

![image](https://user-images.githubusercontent.com/727292/114595165-fa5e4e80-9c85-11eb-8185-d78178f396bc.png)

It's quite simple to add remove at the current stage of the yaml (with matrix), so just let me know what is the call.

Reply:
I completely understand your perspective here 👍 
If you are fine with it then let's remove `full-marian-slim`.